core/src/main/scala/cats/collections/Diet.scala

```diff
@@ -141,21 +144,27 @@ sealed abstract class Diet[A] {
     }
   }
 
-  /**
-    * remove a range from Diet
-    */
-  def removeRange(range: Range[A])(implicit enum: Discrete[A], order: Order[A]): Diet[A] = this match {
-    case EmptyDiet()             => this
+  private def removeRangeIncreasing(range: Range[A])(implicit enum: Discrete[A], order: Order[A]): Diet[A] = this match {
+    case EmptyDiet() => this
     case DietNode(rng, l, r)  =>
-      val left = if(order.lt(range.start, rng.start)) l.removeRange(range) else l
-      val right = if(order.gt(range.end, rng.end)) r.removeRange(range) else r
-        (rng - range) match {
-        case None => merge(l, r)
+      val left = if(order.lt(range.start, rng.start)) l.removeRangeIncreasing(range) else l
+      val right = if(order.gt(range.end, rng.end)) r.removeRangeIncreasing(range) else r
+      rng - range match {
+        case None => merge(left, right)
         case Some((m, None)) => DietNode(m, left, right)
-        case Some((m, Some(n))) => merge(DietNode(m, l, EmptyDiet()), DietNode(n, EmptyDiet(), r))
+        case Some((m, Some(n))) => merge(DietNode(m, left, EmptyDiet()), DietNode(n, EmptyDiet(), right))
       }
   }
 
+  /**
+    * remove a range from Diet
+    */
+  def removeRange(range: Range[A])(implicit enum: Discrete[A], order: Order[A]): Diet[A] =
+    if (order.lteqv(range.start, range.end))
+      removeRangeIncreasing(range)
+    else
+      removeRangeIncreasing(range.reverse)
```

Comment:
This is surprising to me. I would have thought a range with end < start would be an empty Range. Is this not the case?

Reply:
I guess this is a property of Range I didn’t know. I guess there are some reasons for it (maybe it makes map on Range more useful, or maybe just more confusing, not sure which).

Reply:
The `Range` semantics admits `start > end` and then walks backwards (see the sources). The tests produce such ranges and apparently it works everywhere.